### DISCUSSION TITLE
Meta file/format to describe a PDK?

### DISCUSSION THREAD

**Discussion by nrichaud** [The-OpenROAD-Project/OpenROAD#3731] [General]
Hello everyone,
I am new to the project but I have some experience with with PDK and enablement PDK in design flow.

I was wondering how is structured the interface between the PDK file and the flow.

>Do you know if there is a kind of **meta file/format** to describe the PDK and maybe "explain" to the flow where to pick the right file/tools?
Like a kind of python `requirement.txt` file so it could be easy to integrate new PDK and switch from one technology to an other?

Today, my impression is that it is still kind of challenging to have all the pieces or components of a PDK working together with the right version and right place to pick the files.

Thanks

**Comment by rovinski** [comment]
I don't know of any specific standard for PDK distribution. The main issue is that tools / formats are continuously coming in and out of vogue based on the process technology. Many PDKs may be "incomplete" due to lack of enablement for certain tools.

https://github.com/RTimothyEdwards/open_pdks tries to create a standardized directory structure and format for PDKs, but of course there is effort in taking what the vendor provides and trying to fit it into that model because every PDK vendor is different.

[ORFS](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts) simply uses Make variables to point at the necessary files.

**Comment by maliberty** [comment]
I think it is better to accept the PDK as provided than to mess with it as open_pdks requires.  PDKs should be read-only.

**Comment by instaup09876** [comment]
Great question! The idea of a meta file or format to streamline PDK integration into design flows, similar to a `requirements.txt` in Python, would be incredibly useful. It could simplify version management and file selection, making it easier to switch between technologies. I agree, having all PDK components work seamlessly together can be challenging. [#instaup](https://www.instaupgeek.com/)
